dialogue-clients/src/main/java/com/palantir/dialogue/clients/ReloadingClientFactory.java

```diff
@@ -543,48 +533,37 @@ private ImmutableList<TargetUri> getTargetUris(
         return ImmutableSet.copyOf(targetUris).asList();
     }
 
-    private static ProxySelector proxySelector(Optional<ProxyConfiguration> proxyConfiguration) {
+    static ProxySelector proxySelector(Optional<ProxyConfiguration> proxyConfiguration) {
         return proxyConfiguration.map(ClientConfigurations::createProxySelector).orElseGet(ProxySelector::getDefault);
     }
 
     private Refreshable<InternalDialogueChannel> getInternalDialogueChannel(String serviceName) {
         Preconditions.checkNotNull(serviceName, "serviceName");
         String channelName = ChannelNames.reloading(serviceName, params);
 
-        return configurationForService(serviceName).map(conf -> {
-            Preconditions.checkNotNull(
-                    conf, "Refreshable must not provide a null InternalDialogueChannelConfiguration");
+        return configurationForService(serviceName)
+                .map(InternalDialogueChannelConfiguration::serviceConfiguration)
```

Comment:
This is a hack, we're doing dns resolution, then throwing away the dns resolution results to get back the original ServiceConfiguration. The DNS piece is entirely unnecessary here!

Reply:
resolved